### DISCUSSION TITLE
how can I prevent a client to connect no more to my server

### DISCUSSION THREAD

**Discussion by marcatmadu** [rustdesk/rustdesk#5686] [General]
hi,

I am running my own server under ubuntu and now I want to block a certain id to connect to my rustdesk server. I know the ID of the client to block. So I went into the ubuntu GUI and tried t delete the record for this ID with the command:  delete from peer where id='562542587';   

but it tells me that I cannot delete from a readonly database ...  probably because rustdesk has this db (db_v2.sqlite3) in use.
I killed the rustdesk process but still can not open this db for write/delete.

Anyone now how to block a ,certain ID to connect no more ?

**Comment by rustdesk** [comment]
https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/permissions/

**Reply by marcatmadu** [reply]
I'm not running the Pro version but the open source version on ubuntu server .... so I have no GUI web console ... no browser installed etc ... I prefer to leave it like that.  Any link on the use of  blacklist.txt/blocklist.txt ?

PS: I tried also to surf to the console from another computer in my LAN to http://serverip:21114 but no reaction  (fw blocks it in ubuntu server?).

**Reply by dinger1986** [reply]
21114 wont work without pro

**Comment by SergeyMy** [comment]
use "blacklist.txt" "blocklist.txt" for relay_server (hbbr)

**Reply by marcatmadu** [reply]
there's no such file like blacklist.txt or blocklist.txt on my filesystem

**Reply by SergeyMy** [reply]
Of course they don’t exist, they need to be created! which is clearly indicated by the server relay launch log
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/7fb264c5-78e4-4dda-bea5-45bad136cd9e)

**Reply by RichardKimJH** [reply]
What should "blocklist.txt" and "blacklist.txt" contain?
Is there a manual I can refer to?

**Comment by luckman212** [comment]
@RichardKimJH Did you ever figure this out? I have the same question.

**Comment by marcatmadu** [comment]
it has been a while so I do not remember everything.

I have been experimenting with those 2 files. I remember you had to put 
ID's in it one per line.

I did not found a stable solution. It worked, then not anymore, etc.

I found my own solution by using different dynamic dns records which I 
can delete if necessary so that client cannot use anymore my server.

Marc

On 10/12/2023 23:08, Luke Hamburg wrote:

> @RichardKimJH [1] Did you ever figure this out? I have the same 
> question.
> 
> --
> Reply to this email directly, view it on GitHub [2], or unsubscribe 
> [3].
> You are receiving this because you authored the thread.Message ID: 
> ***@***.***>



Links:
------
[1] https://github.com/RichardKimJH
[2] 
https://github.com/rustdesk/rustdesk/discussions/5686#discussioncomment-7813723
[3]

**Comment by luckman212** [comment]
@RichardKimJH I just cloned the repo and searched the entire codebase. I can't find any mention of `blacklist` or `blocklist` anywhere. It may be a Pro-only feature, since they mention "access control" as one of the features...

**Comment by SergeyMy** [comment]
rustdesk-server/src/relay_server.rs
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/26677578-b2c2-47cb-9ec7-5569ed9f2bb2)

**Comment by luckman212** [comment]
Hmm yeah ok I see, I was looking at the rustdesk repo, not the rustdesk-server repo. I see it now.

**Comment by marcatmadu** [comment]
> Hmm yeah ok I see, I was looking at the rustdesk repo, not the rustdesk-server repo. I see it now.

and? does it work ?

**Comment by luckman212** [comment]
@marcatmadu I haven't tested it yet

**Comment by SergeyMy** [comment]
> @marcatmadu I haven't tested it yet

of course it works, it's the server code :)

PS
the so-called PRO functions are the functions of the relay server and the API server, the client is the same! (sorry for revealing the "secret")
